trieste/acquisition/rule.py

```diff
@@ -174,6 +182,8 @@ def acquire(
         search_space: SP_contra,
         datasets: Mapping[str, Dataset],
         models: Mapping[str, ProbabilisticModel],
+        step_number: int,
+        summary_writer: Optional[tf.summary.SummaryWriter] = None,
```

Comment:
Not sure I understand what you're suggesting. Since rules only expose an acquire method (which is called very optimisation step) it fees more flexible to allow them to choose when/whether to log to tensorboard during that step (e.g. we may want to log the acquisition function mutiple times during the acquire method execution).